red-hat-data-services/opendatahub-operator
# Add ability to restrict namespaces being watched.

- [x] JIRA link(s): [RHODS-2732](https://issues.redhat.com/browse/RHODS-2732)


**Testing:**
 1. Install rhods using https://gitlab.cee.redhat.com/data-hub/olminstall with catalogsource: `quay.io/modh/rhods-operator-live-catalog:1.7.0-rhods-2732` [from quay/modh/rhods-operator-live-catalog](https://quay.io/repository/modh/rhods-operator-live-catalog?tag=1.7.0-rhods-2732&tab=tags)
2. Observe operator memory usage redhat-ods-operator using OpenShift "observe" dashboards
3. Make config changes in `redhat-ods-applications`, `redhat-ods-operator` and make sure the operator reacts appropriately (e.g. change replicas of deployment to 2 and watch it get reset by the operator).
4. Create 1000+ namespaces
```bash
for i in {1..1000}
do
   oc create ns dummyns${i}
done
```
Memory should not increase with namespaces
5. Uninstall rhods using the proper `managed-odh` config map with labels `api.openshift.com/addon-managed-odh-delete: 'true'`
